### ISSUE TITLE
Dynamically changing cpPosition

### ISSUE THREAD

**Issue by zainlateef** [zefoy/ngx-color-picker#129]
Is there a way to dynamically change the cpPosition? If I have the cpPosition attribute binded to a variable like so ( [cpPosition]="foo" ), the position of the color picker does not change dynamically with the value of foo. It is only after you reopen the color picker that the position changes. Is this the expected behavior?

**Comment by sconix** [comment]
Yes its expected behavior. Why do you need to change it when dialog is open? You could call close / open for the dialog to update the position after the position change.

**Comment by zainlateef** [comment]
Isn’t that the point of data binding and Angular in general? Whenever I see a property binding such as cpPosition, I automatically assume it will dynamically change when the value it’s assigned to changes. I believe that behavior is universal throughout the Angular architecture, unless I’m mistaken. 
My particular use case is detecting if the color pickers bounding box is outside of the viewport. If it is outside of the viewport, I would flip the box and change cpPosition to “top” so that it reorients itself into the viewport. I will try using the close open dialog methods you suggested. 
Thank you for your timely response!

**Comment by sconix** [comment]
No its not universal it always depends on the component. Updating something dynamically does not always make sense specially performance wise.

The dynamic positioning of the dialog based on the view area is a feature that is coming when I have time to implement it or when somebody provides PR. So the current cpPosition is basically cpDefaultPosition its just named poorly. So it could be made dynamic, but I rather add built in support for "smart" positioning of the dialog.
